### PR TITLE
Dialog height responsive on mobile, avoid css overwrite with important

### PR DESCRIPTION
**Pull request by edellucien**
# How to

Dialog height responsive on mobile, avoid css overwrite with important
# Others

- [x] It's not a hack or at least an unauthorized hack :).
- [x] The images added are optimized.
- [x] Everything in ticket description has been fixed.
- [x] The author of the MR has made its own review before assigning the reviewer.
- [x] IE11 tested.
- [x] The title means something for a human being.
- [x] The title contains [WIP] if it's necessary.
- [x] Labels applied. if it's a release? a hotfix?
- [ ] Tests added.

### CODE REVIEW

File: src/__mocks__/react-i18next.js [hunk 1]
```diff
@@ -4,13 +4,13 @@ const i18n = {
   changeLanguage: jest.fn(() => {}),
 };
 
-const hasChildren = (node) =>
+const hasChildren = node =>
```
Comment (oterral) [inline src/__mocks__/react-i18next.js:7]: rerun `yarn format `

File: src/components/MainDialog/MainDialog.js [hunk 2]
```diff
@@ -37,6 +40,7 @@ const MainDialog = () => {
       <Dialog
         isModal
         name="legallines"
+        wrapperClassName="wkp-dialog-modal"
```
Comment (oterral) [inline src/components/MainDialog/MainDialog.js:43]: we should have to do this . wkp-dialog and wkp.modal have total different css.

File: src/components/Dialog/Dialog.scss [hunk 3]
```diff
@@ -73,43 +117,72 @@
...
       }
     }
   }
 
-  .tm-modal {
-    .tm-dialog-container {
-      width: calc(100% - 40px);
-    }
+  .wkp-dialog-modal {
+    pointer-events: all;
```
Comment (oterral) [inline src/components/Dialog/Dialog.scss:186]: why not on .wkp-modal

Comment (oterral) [reply src/components/Dialog/Dialog.scss:186]: and it seem suseless

File: src/components/Dialog/Dialog.scss [hunk 4]
```diff
@@ -73,43 +117,72 @@
...
-        height: 100%;
-        width: 100%;
-        margin: 0;
+    display: block;
+    position: absolute;
+    bottom: 0;
+    left: 0;
+    right: 0;
+    z-index: 10;
+    height: 100%;
```
Comment (oterral) [inline src/components/Dialog/Dialog.scss:151]: seems useless

File: src/components/Dialog/Dialog.scss [hunk 5]
```diff
@@ -73,43 +117,72 @@
...
+    pointer-events: none;
+    margin: auto;
```
Comment (oterral) [inline src/components/Dialog/Dialog.scss:153]: seems useless

File: src/components/Dialog/Dialog.scss [hunk 6]
```diff
@@ -73,43 +117,72 @@
...
+
+    .tm-dialog {
+      display: block;
+      position: absolute;
```
Comment (oterral) [inline src/components/Dialog/Dialog.scss:157]: seems useless

Comment (oterral) [inline src/components/Dialog/Dialog.scss:157]: seems useless, at least on chrome

File: src/components/Dialog/Dialog.scss [hunk 7]
```diff
@@ -73,43 +117,72 @@
...
+      bottom: 0;
+      left: 0;
+      right: 0;
+      z-index: 10;
+      height: 100%;
+      pointer-events: none;
+      margin: auto;
```
Comment (oterral) [inline src/components/Dialog/Dialog.scss:164]: seems useless , at least on chrome

File: src/components/Dialog/Dialog.scss [hunk 8]
```diff
@@ -73,43 +117,72 @@
   }
 }
 
-.tm-w-m {
-  .tm-dialog {
-    left: 350px;
+.tm-h-m {
```
Comment (oterral) [inline src/components/Dialog/Dialog.scss:120]: Move this class def just after .tm-h-xs

File: src/components/Dialog/Dialog.scss [hunk 9]
```diff
@@ -1,4 +1,8 @@
 .wkp-dialog {
+  .tm-rnd {
+    display: flex;
```
Comment (oterral) [inline src/components/Dialog/Dialog.scss:3]: useless becaus it's overrided by the style attribute

Comment (edellucien) [reply src/components/Dialog/Dialog.scss]: Lint won't accept that, for descending specificity

File: src/components/Dialog/Dialog.scss (hunk 8 above)
Comment (edellucien) [reply src/components/Dialog/Dialog.scss:117]: Lint won't accept that, for descending specificity

**Comment by edellucien** [comment]
Thanks a lot for the feedback. Indeed some useless style because i switch to max-hight based on screen-size to fix on IE.
Code updated.

File: src/components/Dialog/Dialog.scss (hunk 8 above)
Comment (oterral) [reply src/components/Dialog/Dialog.scss:117]: if lint said that

**Comment by edellucien** [comment]
You can test the changes :)

File: src/components/TopicElements/TopicElements.js [hunk 10]
```diff
@@ -151,12 +153,24 @@ function TopicElements({ history, appBaseUrl, staticFilesUrl }) {
 
   const appElements = getComponents(appComponents, elements);
 
+  const onResize = (entries) => {
+    const [entry] = entries;
+    const rect = entry.contentRect;
+    const { width } = rect;
+    // tm-h-xs and tm-w-s
+    dispatch(setIsMobileWidth(width < 768));
```
Comment (oterral) [inline src/components/TopicElements/TopicElements.js:161]:
```suggestion
    dispatch(setIsMobileWidth(width <= 768));
```

File: src/components/TopicElements/TopicElements.js [hunk 11]
```diff
@@ -151,12 +153,24 @@ function TopicElements({ history, appBaseUrl, staticFilesUrl }) {
 
   const appElements = getComponents(appComponents, elements);
 
+  const onResize = (entries) => {
+    const [entry] = entries;
+    const rect = entry.contentRect;
+    const { width } = rect;
+    // tm-h-xs and tm-w-s
```
Comment (oterral) [inline src/components/TopicElements/TopicElements.js:160]:
```suggestion
    // tm-h-xs
```

Comment (oterral) [inline src/components/TopicElements/TopicElements.js:160]:
```suggestion
    // tm-w-s
```

**Pull request merged**